edge-util/src/Microsoft.Azure.Devices.Edge.Util/Proxy.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Net;
+    using Microsoft.Extensions.Logging;
+
+    public static class Proxy
+    {
+        public static Option<IWebProxy> Parse(string proxyUri, ILogger logger)
+        {
+            if (string.IsNullOrWhiteSpace(proxyUri))
+            {
+                return Option.None<IWebProxy>();
+            }
+
+            var uri = new Uri(proxyUri);
+            if (string.IsNullOrEmpty(uri.UserInfo))
+            {
+                // http://proxyserver:1234
+                logger.LogInformation($"Detected proxy {uri}");
+                return Option.Some<IWebProxy>(new WebProxy(uri));
+            }
+            else
+            {
+                string[] parts = uri.UserInfo.Split(':');
+                var credentials = new NetworkCredential
+                {
+                    UserName = Uri.UnescapeDataString(parts[0])
+                };
+
+                if (parts.Length == 1)
+                {
+                    // http://user@proxyserver:1234
+                    logger.LogInformation($"Detected proxy {uri}");
+                }
+                else
+                {
+                    // http://user:password@proxyserver:1234
+                    credentials.Password = Uri.UnescapeDataString(parts[1]);
+
+                    // log the proxy URI without the password
+                    int pos = uri.ToString().IndexOf(uri.UserInfo, StringComparison.InvariantCulture);
+                    int begin = pos + parts[0].Length;
+                    int count = parts[1].Length + 1;
+                    logger.LogInformation($"Detected proxy {uri.ToString().Remove(begin, count)}");
```

Comment:
Nit: It feels better to replace password with ***, instead of just removing it..

Reply:
Then you can simply use string.Replace..

Reply:
Makes sense. I'll do that.

Reply:
Password now replaced with ***, but didn't use Replace since it replaces every occurrence. Maybe it's a long shot, but a poor password might cause more than one part of the string to get starred, and then everyone who saw the logs would know the password. Too bad too, since that code would have been much simpler...

Reply:
Emm, just a suggestion - can we not use something like this - 
uri.ToString().Replace($":{password}", ":*****);

Reply:
Alright, done.